packages/algob/src/internal/cli/cli.ts

```diff
@@ -43,94 +43,170 @@ function ensureValidNodeVersion (packageJson: PackageJson): void {
   }
 }
 
-/* eslint-disable sonarjs/cognitive-complexity */
-async function main (): Promise<void> {
+function printErrRecur (error: BuilderError): void {
+  console.error(error.stack);
+  if (error.parent) {
+    if (error.parent instanceof BuilderError) {
+      printErrRecur(error.parent);
+    } else {
+      console.error(error.parent);
+    }
+  }
+}
+
+function printStackTraces (showStackTraces: boolean, error: BuilderError): void {
+  if (showStackTraces) {
+    printErrRecur(error);
+  } else {
+    console.error(`For more info run ${ALGOB_NAME} with --show-stack-traces or add --help to display task-specific help.`);
+  }
+}
```

Comment:
let's move this `print*` functions to other module.

Reply:
This is out of scope of this task.

Reply:
> This is not a valid typescript syntax:

I know - it's a pseudocode, but you see the idea.

Reply:
well, this was introduced in this PR, so it is part of this PR.

Reply:
It was moved in this PR, not introduced here.